### DISCUSSION TITLE
Copilot has no context anymore on VS-code, it's almost completely useless by now

### DISCUSSION THREAD

**Discussion by cmp-nct** [community/community#53057] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Look at the below code which is just a bit more than 1000 tokens of context.
When Github Copilot launched it was clear that Copilot could see a large part of my code, it properly suggested code based on context.
Over the past months this was more and more dumbed down. It is based on Codex so I guess it was launched with 8000 tokens context.
The below code is just 1000 tokens, it's hard to tell where exactly Copilot is cut off but it has no clue about the function.
When I originally subscribed copilot wouldn't be so clueless to suggest (1,2,3,4) as arguments.

I'll look into CodeWhisperer if it's better than that, will update here.
**Update**: CodeWhisperer has 2 thousand of previous tokens in it's context (based on GPT-3 token count).
Copilot seems to be hard-cut at 1 thousand tokens.
In my opinon both is low but 1k is not even enough to cover one medium function.
Would be awesome to get a bump if this is going to be fixed, right now Copilot is 10 USD/month and CodeWhisperer is free.

```
struct ggml_tensor * get_random_tensor(
        struct ggml_context * ctx0,
        int ndims,
        int64_t ne[],
        float fmin,
        float fmax) {
    struct ggml_tensor * result = ggml_new_tensor(ctx0, GGML_TYPE_F32, ndims, ne);

    switch (ndims) {
        case 1:
            for (int i0 = 0; i0 < ne[0]; i0++) {
                ((float *)result->data)[i0] = frand()*(fmax - fmin) + fmin;
            }
            break;
        case 2:
            for (int i1 = 0; i1 < ne[1]; i1++) {
                for (int i0 = 0; i0 < ne[0]; i0++) {
                    ((float *)result->data)[i1*ne[0] + i0] = frand()*(fmax - fmin) + fmin;
                }
            }
            break;
        case 3:
            for (int i2 = 0; i2 < ne[2]; i2++) {
                for (int i1 = 0; i1 < ne[1]; i1++) {
                    for (int i0 = 0; i0 < ne[0]; i0++) {
                        ((float *)result->data)[i2*ne[1]*ne[0] + i1*ne[0] + i0] = frand()*(fmax - fmin) + fmin;
                    }
                }
            }
            break;
        case 4:
            for (int i3 = 0; i3 < ne[3]; i3++) {
                for (int i2 = 0; i2 < ne[2]; i2++) {
                    for (int i1 = 0; i1 < ne[1]; i1++) {
                        for (int i0 = 0; i0 < ne[0]; i0++) {
                            ((float *)result->data)[i3*ne[2]*ne[1]*ne[0] + i2*ne[1]*ne[0] + i1*ne[0] + i0] = frand()*(fmax - fmin) + fmin;
                        }
                    }
                }
            }
            break;
        default:
            assert(false);
    };

    return result;
}

int main(void) {
    struct ggml_tensor *t1 = get_random_tensor(... <- it recommends me to use 1,2,3,4
```

**Comment by KingOfPeru** [comment]
I agree it cant even do // add two numbers . Everyday becoming more useless.
